Source/ASTextNode.mm

```diff
@@ -110,6 +110,7 @@ - (BOOL)isEqual:(ASTextNodeRendererKey *)object
  dispatch_once(&onceToken, ^{
    __rendererCache = [[NSCache alloc] init];
    __rendererCache.countLimit = 500; // 500 renders cache
```

Comment:
I know this is for ASTextNode1...however, this might actually be quite a bit of memory. Do we dump this cache on memory warning? Is there anything simple we can do to make it more responsive to some combination of memory consumption (by objects in the cache), device memory size, or frequency of memory pressure?

I think there's a need for something like ASMemoryManager (or MemoryOverlord lol) that provides some conveniences around memory. Total memory capacity could be bucketed into enums with perhaps a low/med/high — especially iPads would mostly be in low or med, for example. Most valuably would be registering some blocks or observers to call upon memory pressure, and providing an indication of how severe or sustained the memory pressure is.

For example, repeated memory pressure should certainly reduce the countLimit, display range size, or other such parameters rather than just one-time cache dumps. Even if there's no mechanism to increase these again, this would allow us to choose more aggressive defaults without as much concern for apps that may have e.g. many large text blocks.

Reply:
Filed a tracking task for this idea: https://github.com/TextureGroup/Texture/issues/927

Reply:
A memory management object is really interesting. Take a look at this, which provides 3 levels of severity. This is the hook that NSCache uses. Side note, I'm in favor of removing the explicit count limit here and letting NSCache decide.

https://developer.apple.com/documentation/dispatch/dispatch_source_memory_pressure_event_flags?language=objc